jaeger-jaxrs2/src/main/java/com/uber/jaeger/filters/jaxrs2/ClientFilter.java

```diff
@@ -36,30 +33,22 @@
   private final Tracer tracer;
   private final TraceContext traceContext;
 
+  private final ClientSpanCreationFilter spanCreationFilter;
+  private final ClientSpanInjectionFilter spaninjectionFilter;
+
   public ClientFilter(Tracer tracer, TraceContext traceContext) {
     this.tracer = tracer;
     this.traceContext = traceContext;
+
+    this.spanCreationFilter = new ClientSpanCreationFilter(tracer, traceContext);
+    this.spaninjectionFilter = new ClientSpanInjectionFilter(tracer, traceContext);
   }
 
   @Override
   public void filter(ClientRequestContext clientRequestContext) throws IOException {
```

Comment:
Just so I follow, for your use case, you'll extend the ClientFilter and then override this filter function to add your baggage between the spanCreationFilter and spanInjectionFilter?

What do we do in the future when other people want to add extra functionality? Will they add their functionality into your filter?

Reply:
then someone can add a CompositeFilter

Reply:
right, so is that the plan?

Reply:
there's no plan - it's not a problem today, and it's solvable in the future with composite filter, so we don't need to over-engineer today for non-existing need